src/components/checkbox/Checkbox.js

```diff
@@ -173,6 +173,7 @@ export class CheckBoxComponent extends BaseComponent {
     if (!this.input) {
       return;
     }
+    this.data[this.component.key] = value;
```

Comment:
Why did you need to add this? This is done within ```updateValue``` which is called at the bottom of this method. That is also used to determine if the value changed so it can trigger an update. I am pretty sure this breaks the "changed" functionality where the change event is no longer fired on checkboxes after doing this.

Reply:
I can take another look at this but what happens is when the data is cleared on "clear on hide" it no longer exists in this.data. This was causing lots of issues where the component was attempting to access a property that no longer existed. I'll try taking this out in several places and see if it still passes tests.